### PR TITLE
node homework w2 done

### PR DESCRIPTION
**Pull request by ahmetcgl83**

### CODE REVIEW

File: homework/hackyourtemperature/app.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+app.use(express.json());
+
+app.get("/", (req, res) => {
+	res.send("hello from backend to frontend!");
+});
+
+app.post("/weather/:cityName", async (req, res) => {
+	const cityName = req.params.cityName;
+
+	const response = await fetch(
```
Comment (MahmoudMito) [inline homework/hackyourtemperature/app.js:14]: The code looks good but Its throwing an error at fetch as it is imported/required at the top of the code, so wasn't able to run it

File: homework/hackyourtemperature/app.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+const app = express();
+app.use(express.json());
+
+app.get("/", (req, res) => {
+	res.send("hello from backend to frontend!");
+});
+
+app.post("/weather/:cityName", async (req, res) => {
+	const cityName = req.params.cityName;
```
Comment (MahmoudMito) [inline homework/hackyourtemperature/app.js:12]: passing the name of the city in the url will work , however when using POST using the body of the request makes more sense as you can pass several attributes if needed in the body.

File: homework/hackyourtemperature/__test__/app.test.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import app from "../app.js";
+import supertest from "supertest";
+const request = supertest(app);
+
```
Comment (MahmoudMito) [inline homework/hackyourtemperature/__test__/app.test.js:4]: Tests looks good and I like that you also testing the negative cases, but It failed because of the app.js error

File: homework/hackyourtemperature/__test__/app.test.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+		});
+
+		it("should respond with a 400 status code if city name is undefined", async () => {
+			const response = await request.post(
+				"/weather/gibberishCityName1234567890"
+			);
+			expect(response.statusCode).toBe(400);
+		});
+
+		it("should respond with a 404 status code if there is no city name", async () => {
```
Comment (MahmoudMito) [inline homework/hackyourtemperature/__test__/app.test.js:26]: I don't see the 404 response in app.js so it will fail here, please adjust the app.js to include the 404 response

**Review by MahmoudMito — CHANGES_REQUESTED** [review]

File: homework/hackyourtemperature/__test__/app.test.js [hunk 5]
```diff
@@ -0,0 +1,31 @@
+import app from "../app.js";
+import supertest from "supertest";
+const request = supertest(app);
+
+describe("POST /weather/:cityName", () => {
```
Comment (MahmoudMito) [inline homework/hackyourtemperature/__test__/app.test.js:5]: Test method is failing, you changed the city location from the url to the body and ddin't reflect that in the test.

There are several failures in the status code not meeting the expected values.

image


image

**Review by MahmoudMito — COMMENTED** [review]
Please adjust the test methods as they are failing

File: homework/hackyourtemperature/__test__/app.test.js (hunk 5 above)
Comment (MahmoudMito) [reply homework/hackyourtemperature/__test__/app.test.js:5]: Test classes are now working and got all of the tests passing

**Review by MahmoudMito — APPROVED** [review]
Thanks for doing the changes.

test classes are working & API are now working
